src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1150,6 +1191,76 @@ InitializeWorkerNodeCache(void)
 
 
 /*
+ * GetLocalGroupId returns the group identifier of the local node. The function assumes
+ * that pg_dist_local_node_group has exactly one row and has at least one column.
+ * Otherwise, the function errors out.
+ */
+int
+GetLocalGroupId(void)
+{
+	SysScanDesc scanDescriptor = NULL;
+	ScanKeyData scanKey[1];
+	int scanKeyCount = 0;
+	HeapTuple heapTuple = NULL;
+	TupleDesc tupleDescriptor = NULL;
+	Oid groupId = InvalidOid;
+	Relation pgDistLocalGroupId = NULL;
+	bool invalidationRegistered = false;
+
+	/*
+	 * Already set the group id, no need to read the heap again. Also, note that
+	 * the call to heap_open() below needs to be run only when transaction in progress.
```

Comment:
I don't really get the "the call to heap_open() needs to be run only when transaction in progress" part, maybe remove it.